cmd/util/cmd/execution-state-extract/cmd.go

```diff
@@ -78,8 +86,49 @@ func run(*cobra.Command, []string) {
 		mappings[k] = v
 	}
 
-	err = extractExecutionState(flagExecutionStateDir, stateCommitment, flagOutputDir, log.Logger, mappings)
+	fixedMappings, err := FixMappings(mappings)
+	if err != nil {
+		log.Fatal().Err(err).Msg("cannot fix mappings")
+	}
+
+	err = extractExecutionState(flagExecutionStateDir, stateCommitment, flagOutputDir, log.Logger, fixedMappings)
 	if err != nil {
 		log.Fatal().Err(err).Msg("cannot generate checkpoint with state commitment")
 	}
 }
+
+func fullKey(owner, controller, key string) string {
+	// https://en.wikipedia.org/wiki/C0_and_C1_control_codes#Field_separators
+	return strings.Join([]string{owner, controller, key}, "\x1F")
+}
+
+func fullKeyHash(owner, controller, key string) string {
+	hasher := hash.NewSHA2_256()
+	return string(hasher.ComputeHash([]byte(fullKey(owner, controller, key))))
+}
+
+func FixMappings(mappings map[string]delta.Mapping) (map[string]delta.Mapping, error) {
```

Comment:
Could you add some comments to explain why the mapping needs to be fixed?

Reply:
Done!